Source/WebCore/loader/SubframeLoader.cpp

```diff
@@ -305,7 +305,11 @@ RefPtr<LocalFrame> FrameLoader::SubframeLoader::loadSubframe(HTMLFrameOwnerEleme
     ReferrerPolicy policy = ownerElement.referrerPolicy();
     if (policy == ReferrerPolicy::EmptyString)
         policy = document->referrerPolicy();
-    String referrerToUse = SecurityPolicy::generateReferrerHeader(policy, url, referrer, OriginAccessPatternsForWebProcess::singleton());
+    // For any new (about:blank) browsing context, step 16 of
+    // https://html.spec.whatwg.org/#creating-a-new-browsing-context requires
+    // setting the referrer to "the serialization of creator's URL" — that is,
+    // the full URL, without regard to Referrer Policy.
+    auto referrerToUse = url == aboutBlankURL() ? referrer : SecurityPolicy::generateReferrerHeader(policy, url, referrer, OriginAccessPatternsForWebProcess::singleton());
```

Comment:
I think we’d want to use an `isAboutBlank()` function instead of `== aboutBlankURL()`, if we have such a thing.

But also, why wouldn’t this logic be inside `SecurityPolicy::generateReferrerHeader` instead of here?

Reply:
Thanks — replaced the `url == aboutBlankURL()` with `url.isAboutBlank()`, and moved the logic directly into `SecurityPolicy::generateReferrerHeader`

Reply:
>  moved the logic directly into `SecurityPolicy::generateReferrerHeader`

…and now I’ve moved it back into `FrameLoader::SubframeLoader::loadSubframe`…

> why wouldn’t this logic be inside `SecurityPolicy::generateReferrerHeader` instead of here?

I rightly should have realized the answer to that when you first asked — but it’s been a couple months since I initially wrote the patch, and I didn’t actually recall it until my memory was jogged by seeing the test at https://github.com/web-platform-tests/wpt/blob/master/html/browsers/browsing-the-web/navigating-across-documents/source/navigate-child-src-about-blank.html start failing after I moved the logic into `SecurityPolicy::generateReferrerHeader`.

So the answer is that per step 16 of https://html.spec.whatwg.org/multipage/#creating-a-new-browsing-context, this should _only_ happen when creating a new browsing context (so, in `FrameLoader::SubframeLoader::loadSubframe`) — and per step 13 of https://html.spec.whatwg.org/multipage/#initialise-the-document-object, should _not_ happen when creating and initializing a new `Document` object (in which case, Referrer Policy _is_ applied, and so, `SecurityPolicy::generateReferrerHeader` is used).

(https://bugs.webkit.org/show_bug.cgi?id=242965#c4 has a related more-detailed explanation of the spec requirements.)

So, along with keeping the logic in `FrameLoader::SubframeLoader::loadSubframe`, I expanded my code comment there to include an explicit explanation about why the logic’s not in `SecurityPolicy::generateReferrerHeader`. But if the verbosity of that comment seems like overkill, please lemme know, and I can try to trim it back a bit.